org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/semantictokens/SemanticTokensVisitor.java

```diff
@@ -18,204 +18,212 @@
+		this.manager = manager;
+		this.document = document;
+		this.tokens = new ArrayList<>();
+	}
+
+	private class SemanticToken {
+		private final TokenType tokenType;
+		private final TokenModifier[] tokenModifiers;
+		private final int offset;
+		private final int length;
+
+		public SemanticToken(int offset, int length, TokenType tokenType, TokenModifier[] tokenModifiers) {
+			this.offset = offset;
+			this.length = length;
+			this.tokenType = tokenType;
+			this.tokenModifiers = tokenModifiers;
+		}
+
+		public TokenType getTokenType() {
+			return tokenType;
+		}
+
+		public TokenModifier[] getTokenModifiers() {
+			return tokenModifiers;
+		}
+
+		public int getOffset() {
+			return offset;
+		}
+
+		public int getLength() {
+			return length;
+		}
+	}
+
+	public SemanticTokens getSemanticTokens() {
+		List<Integer> data = encodeTokens();
+		return new SemanticTokens(data);
+	}
+
+	private List<Integer> encodeTokens() {
+		List<Integer> data = new ArrayList<>();
+		int currentLine = 0;
+		int currentColumn = 0;
+		for (SemanticToken token : this.tokens) {
+			int[] lineAndColumn = JsonRpcHelpers.toLine(this.document, token.getOffset());
+			int line = lineAndColumn[0];
+			int column = lineAndColumn[1];
+			int deltaLine = line - currentLine;
+			if (deltaLine != 0) {
+				currentLine = line;
+				currentColumn = 0;
+			}
+			int deltaColumn = column - currentColumn;
+			currentColumn = column;
+			// Disallow duplicate/conflict token (if exists)
+			if (deltaLine != 0 || deltaColumn != 0) {
+				int tokenTypeIndex = manager.getTokenTypes().indexOf(token.getTokenType());
+				TokenModifier[] modifiers = token.getTokenModifiers();
+				int encodedModifiers = 0;
+				for (TokenModifier modifier : modifiers) {
+					int bit = manager.getTokenModifiers().indexOf(modifier);
+					encodedModifiers = encodedModifiers | (0b00000001 << bit);
+				}
+				data.add(deltaLine);
+				data.add(deltaColumn);
+				data.add(token.getLength());
+				data.add(tokenTypeIndex);
+				data.add(encodedModifiers);
+			}
+		}
+		return data;
+	}
+
+	/**
+	* Adds a semantic token to the list of tokens being collected by this
+	* semantic token visitor.
+	*
+	* @param node The AST node representing the location of the semantic token.
+	* @param tokenType The type of the semantic token.
+	* @param modifiers The modifiers of the semantic token.
+	*
+	* @apiNote This method is order-dependent because of {@link #encodeTokens()}.
+	* If semantic tokens are not added in the order they appear in the document,
+	* the encoding algorithm might discard them.
+	*/
+	private void addToken(ASTNode node, TokenType tokenType, TokenModifier[] modifiers) {
+		int offset = node.getStartPosition();
+		int length = node.getLength();
+		SemanticToken token = new SemanticToken(offset, length, tokenType, modifiers);
+		tokens.add(token);
+	}
+
+	@Override
+	public boolean visit(SimpleName node) {
+		TokenType tokenType = TokenType.getApplicableToken(node.resolveBinding());
```

Comment:
https://github.com/redhat-developer/vscode-java/issues/1434 is introduced again. And by the way it's not only for `java`. 
![image](https://user-images.githubusercontent.com/2351748/87116406-5388f400-c2a8-11ea-9e8d-d0b3d17d767b.png)


![image](https://user-images.githubusercontent.com/2351748/87116297-0573f080-c2a8-11ea-9251-b82db81c2b52.png)